File: iosgl/iosgl/bt/display/particle/CombineParticle.m
//
//  CombineParticle.m
//  iosgl
//
//  Created by zhao on 23/2/2020.
//  Copyright © 2020 zhao. All rights reserved.
//

#import "CombineParticle.h"
#import "Display3DParticle.h"
#import "Display3DLocusPartilce.h"
#import "IBind.h"


@interface CombineParticle ()



@property (nonatomic, assign)float  _time;
@property (nonatomic, assign)BOOL  _isInGroup;
@property (nonatomic, strong)Vector3D*  _groupPos;
@property (nonatomic, strong)Vector3D*  _groupRotation;
@property (nonatomic, strong)Vector3D*  _groupScale;
@property (nonatomic, strong)id<IBind>  _bindTarget;

//@property(assign,nonatomic) id<DynamicBaseCellDelegate>delegate;

@end

@implementation CombineParticle

-(id<IBind>)bindTarget;
{
    return __bindTarget;
}
-(void)setBindTarget:(id<IBind>)value;
{
    __bindTarget=value;
}
-(void)setX:(float)value; {
    self.bindVecter3d.x=value;
}
-(float)x;{
    return  self.bindVecter3d.x;
}
-(void)setY:(float)value;{
    self.bindVecter3d.y=value;
}
-(float)y;{
    return self.bindVecter3d.y;
}
-(void)setZ:(float)value;{
    self.bindVecter3d.z=value;
}
-(float)z;{
    return self.bindVecter3d.z;
}
 
-(void)setScaleX:(float)value;{
   self.bindScale.x=value;
  
}
-(float)scaleX;{
    return self.bindScale.x;
}
-(void)setScaleY:(float)value;{
   self.bindScale.y=value;
      
}
-(float)scaleY;{
    return self.bindScale.y;
}
-(void)setScaleZ:(float)value;{
     self.bindScale.z=value;
      
}
-(float)scaleZ;{
    return self.bindScale.z;
}


- (instancetype)init
{
    self = [super init];
    if (self) {
        self._displayAry=[[NSMutableArray alloc]init];
        self._time=0;
        self.bindVecter3d =[[Vector3D alloc]init];
        self.bindScale = [[Vector3D alloc]x:1 y:1 z:1];
        self.bindMatrix = [[Matrix3D alloc]init];
        self.invertBindMatrix =  [[Matrix3D alloc]init];
        self.groupMatrix =  [[Matrix3D alloc]init];
        self.groupRotationMatrix = [[Matrix3D alloc]init];
    }
    return self;
}

-(void)addPrticleItem:(Display3DParticle*)dic
{
    dic.visible=YES;
    [dic setBind:self.bindVecter3d rotation:self.bindMatrix scale:self.bindScale invertRotation:self.invertBindMatrix groupMatrix:self.groupMatrix];
    [self._displayAry addObject:dic];
    
}
-(void)updateTime:(float)t;
{
    self._time += t;
    for(int i=0;i<self._displayAry.count;i++)
    {
        [self._displayAry[i] updateTime:self._time];
    }
}
-(void)reset;
{
    self._time=0;
    for (int i = 0; i < self._displayAry.count; i++) {
        [self._displayAry[i] reset];
    }
    
}
-(void)update;
{
    for(int i=0;i<self._displayAry.count;i++)
    {
        self._displayAry[i].scene3d=self.scene3d;
        
        if([ self._displayAry[i] isKindOfClass:[Display3DLocusPartilce class ] ]){
           
        }
        [self._displayAry[i] update];
        
    }
}
-(void)updateItem:(int)idx;
{
    
}

@end

File: android/two/app/src/main/java/z3d/display/particle/CombineParticle.java
package z3d.display.particle;

import android.util.Log;

import java.util.ArrayList;
import java.util.List;

import z3d.vo.Matrix3D;
import z3d.vo.Vector3D;

public class CombineParticle {
    public CombineParticleData sourceData;
    public String url;
    private List<Display3DParticle> displayAry;
    public float _time;
    public float maxTime;
    public int type; //类型
    public Matrix3D bindMatrix;
    public Vector3D bindVecter3d;
    public Vector3D bindScale;
    public Matrix3D invertBindMatrix;
    private String _bindSocket;
    private float _rotationX;
    private float _rotationY;
    private float _rotationZ;
    private boolean _isInGroup;
    private Vector3D _groupPos;
    private Vector3D _groupRotation;
    private Vector3D _groupScale;
    public Matrix3D groupMatrix;
    public Matrix3D groupRotationMatrix;
    public boolean hasMulItem;
    public boolean sceneVisible;
    public boolean dynamic;
    public boolean hasDestory;


    public  CombineParticle()
    {
        this._time=0;
        this.displayAry=new ArrayList<>();
    }

    public void addPrticleItem(Display3DParticle dis) {
        dis.visible = true;
        dis.setBind(this.bindVecter3d, this.bindMatrix, this.bindScale, this.invertBindMatrix, this.groupMatrix);
        this.displayAry.add(dis);

    }
    private String TAG="dd";
    public  void updateTime(float t)
    {


        this._time+=t;

        for(int i=0;i<this.displayAry.size();i++)
        {
            this.displayAry.get(i).updateTime(this._time);
        }
    }

    public  void  upframe()
    {

        for(int i=0;i<this.displayAry.size();i++)
        {
            this.displayAry.get(i).upframe();
        }
    }

}